### ISSUE TITLE
Unclear output shape of dets and labels

### ISSUE THREAD

**Issue by BotaoPeng98** [roboflow/rf-detr#134]
Hi, 

I exported my rf-detr pth model into an ONNX model, but I found the output shape of dets and labels isn't quite understandable for me. I guess it may be with dynamic output shapes.

![Image](https://github.com/user-attachments/assets/70e115e5-2909-412a-a8c9-2136d3e23992)

To be clear, I used onnxslim to slim the ONNX model, and I got this. 

![Image](https://github.com/user-attachments/assets/0978c48d-27a4-4b2d-8350-42e1cab433cc)

The dynamic output shapes became static. I was wondering how this happened.
1. What causes the dynamic output shapes? 
2. Can the static output shapes be calculated from the network's parameters?
3. Static input shapes while dynamic output shapes. Why do that?

Best regards,
Noah

**Comment by isaacrob-roboflow** [comment]
the model outputs 300 detections, as that is the parameter value for num_queries. each detection has a bbox (length 4) and logits per class (length n classes). I'm guessing you have 4 classes

**Comment by BotaoPeng98** [comment]
@isaacrob-roboflow Yep! But I’d like to come back to my questions.

**Comment by isaacrob-roboflow** [comment]
I'm saying that the output from the model is static and can be determined from the config in the way that I described above. It only becomes dynamic if you threshold by confidence.

**Comment by BotaoPeng98** [comment]
Well, I apologize that I haven't made my questions clear.

I have set the config like you mentioned, bbox and 4 classes, so the model output is supposed to be static. While the shapes of model's output which shows in netron is dets: float32[Concatdets_dim_0, Concatdets_dim_1,4] and labels: float32[Addlabels_dim_0, Addlabels_dim1,4]. What confused me is that the Concatdets_dim_0, Concatdets_dim_1 etc. are not static. These signs made me feel like the output might be dynamic.

I am doubtful about how the confidence thresholds the outputs' shapes. If I set the confidence from 0.5 to 0.7, then more bboxes are filtered. But the shapes still are [1, 300, 4] .

Thanks a ton for your kindness and patience.

**Comment by isaacrob-roboflow** [comment]
Confidence filtering happens after the onnx graph. Onnx output is static but if y it filter by confidence you'll see what appears to be dynamic output from the system as a whole.
